File: booking-service/src/main/java/com/thanhtd/aerona/booking/dto/Flight.java
package com.thanhtd.aerona.booking.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class Flight {
    @JsonProperty("flight_id")
    private String flightId;

    @JsonProperty("flight_code")
    private String fightCode;

    @JsonProperty("departure")
    private AirportLocation departure;

    @JsonProperty("arrival")
    private AirportLocation arrival;

    @JsonProperty("duration")
    private Long duration;

    @JsonProperty("price")
    private Float price;

    private String currency;

    private String airline;

    @JsonProperty("seats_available")
    private Integer seatsAvailable;

    private Integer stops;

    @JsonProperty("stop_details")
    private List<StopDetail> stopDetails;

    @JsonProperty("airplane_type")
    private String airplaneType;

    @JsonProperty("flight_date")
    private String flightDate;

    @JsonProperty("created_at")
    private Date createdAt;

    @JsonProperty("updated_at")
    private Date updatedAt;

    private String status;
}

File: ticket-service/internal/api/dto/flight.go
package dto

import (
	"time"
)

type Flight struct {
	FlightId       string          `json:"flight_id"`
	FlightCode     string          `json:"flight_code"`
	Departure      AirportLocation `json:"departure"`
	Arrival        AirportLocation `json:"arrival"`
	Duration       int64           `json:"duration"`
	Price          float32         `json:"price"`
	Currency       string          `json:"currency"`
	Airline        string          `json:"airline"`
	SeatsAvailable int             `json:"seats_available"`
	Stops          int             `json:"stops"`
	StopDetails    []StopDetail    `json:"stop_details"`
	AirplaneType   string          `json:"airplane_type"`
	FlightDate     string          `json:"flight_date"`
	CreatedAt      *time.Time      `json:"created_at"`
	UpdatedAt      *time.Time      `json:"updated_at"`
	Status         string          `json:"status"`
}

type AirportLocation struct {
	AirportName string     `json:"airport_name"`
	AirportCode string     `json:"airport_code"`
	City        string     `json:"city"`
	Country     string     `json:"country"`
	Scheduled   *time.Time `json:"scheduled"`
}

type StopDetail struct {
	StopId        int        `json:"stop_id"`
	AirportCode   string     `json:"airport_code"`
	DepartureTime *time.Time `json:"departure_time"`
	ArrivalTime   *time.Time `json:"arrival_time"`
}
